### ISSUE TITLE
Drawing

### ISSUE THREAD

**Issue by askhat** [dankuck/vue-easeljs#4]
First I should say thank you for this library. I appreciate your work very much.

I have a question, how do I draw on the canvas. For example, I want to draw a mesh with arbitrary properties using `graphics`. And it is not clear to me how to access an Easel instance so I can supply imperative drawing instructions

**Comment by dankuck** [comment]
Hi askhat, thanks for the kind words.

At this point the library isn't designed to handle drawing. I haven't put a lot of thought into how drawing and the Vue.js component concept could go together, but I can see that they are very different concepts.

If you have any ideas for how it should work, I'm all ears. I'd love to talk through how we could implement it.

The EaselJS library that this is built on top of includes a demo of drawing here: https://www.createjs.com/demos/easeljs/curveto . Click "Live Edit" to see how their demo works.

**Comment by askhat** [comment]
Perhaps my chose of words brought some misconception on our conversation. By drawing I refer not to literate drawing by movement of hand. I actually refer to supplying code with methods called on `context` object, so it render something else than this library's components.

I though of additional event on `EaselCanvas` component, which will contain a payload with `context`. So it can be handled in host component, like so:

```html
<easel-canvas @expose="drawStuff">...</easel-canvas>
```

**Comment by dankuck** [comment]
If you really want to access the context object from outside the canvas element, I'd just attach a `ref` to it and access it that way.

```html
<template>
  <easel-canvas ref="myCanvas">...</easel-canvas>
</template>

<script>
...
  this.context = this.$refs.myCanvas.context;
...
</script>
```

**Comment by dankuck** [comment]
If I was going to upgrade this with more powerful line drawing with curves and fills, etc., I'd have to look into how EaselJS makes that work. I think it would end up looking something like this:

```html
<easel-canvas>
  <easel-shape>
    <easel-point x="10" y="10">
    <easel-point x="20" y="20">
    <easel-point x="10" y="20" curve="5%">
  </easel-shape>
</easel-canvas>
```

The intended meaning would be that each point would connect to the next to produce a shape.

**Comment by askhat** [comment]
So, after tinkering a bit I was able perform my task.

Except few remarks: I found that using `context` provided by `$refs.myCanvas` is useless, because everything drawn on it will be cleared on the next tick. However, it is possible to use `graphics` of a `<easel-shape/>`. Like so:

```javascript
<template>
  <easel-canvas>
    <easel-shape ref="myShape">
  </easel-canvas>
</template>

<script>
methods: {
  draw () {
    const graphics = this.$refs.myShape.component.graphics
    graphics.clear()
    graphics.someDrawingMethod()
  }
}
</script>
```

Binding on `@tick` was not performant enough, so I bound method on some events and watchers. The only shortcoming is that I persistently observe `Error: "No bounds available"` since `<easel-shape/>` has no `form` and `dimensions`. So it would be nice if `<easel-shape/>` would've been more relaxed about its props.

I believe the issue may be closed.
